c10/util/flat_hash_map.h

```diff
@@ -32,9 +32,14 @@ C10_CLANG_DIAGNOSTIC_IGNORE("-Wimplicit-int-float-conversion")
 #endif
 
 #ifndef _MSC_VER
+#if defined(__clang__)
+C10_CLANG_DIAGNOSTIC_PUSH()
+C10_CLANG_DIAGNOSTIC_IGNORE("-Wshadow")
+#else //!__clang__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wshadow"
-#endif
+#endif //!__clang__
+#endif // !_MSC_VER
```

Comment:
Looks like you are removing all shadow violations in this file. In that case, why do you need `-Wshadow` diagnostic ignore?

Reply:
Turns out there were a few more shadow warnings after removing the pragmas... can't say I fully understand that (why some got suppressed but others didn't).

That said, actually fixing the warnings does seem like a good idea - so revising the request with those changes.... now!